design/ceph/multus-network.md

```diff
@@ -90,111 +90,53 @@ interface will be added to the host network namespace for all nodes that will ru
-
-Restarting a node will cause the multus interface in the host namespace to go away. On restart, the
-holder pod will get a new interface, and the mover will copy it into the host networking namespace again.
-
-When a new node is added, holder and mover pods are added to it by their DaemonSets, and the
-move/copy process described above occurs on the node.
-
-The holder and mover DaemonSets should be deleted when the CSI driver components are removed. Both
-the termination of the holder pods as well as the mover pods triggers the mover to remove the multus
-interfaces from the host network namespace of a given node.
-
-The initial implementation of this design will be limited to supporting a single CephCluster with
-Multus until we can be sure that the CSI plugin can support multiple migrated interfaces as well as
-interfaces that are added and removed dynamically. This limitation will be enforced by allowing only
-a single instance of the holder DaemonSet. A possible (future) partial implementation may be
-possible by restarting the CSI plugin Pods when network interfaces are added or removed.
-
-**Known issue:** the Docker container runtime does not use Linux's native `/var/run/netns`
-directory. This mitigation is known to work on cri-o runtime but not Docker. Therefore, this feature
-will be disabled by default and enabled optionally by the
-`ROOK_CSI_MULTUS_USE_HOLDER_MOVER_PATTERN=true` variable in the Rook-Ceph operator's config.
-
-A previous version of the CSI proposal had the holder Pods creating "setup" and "teardown"
-Kubernetes Jobs for migrating/un-migrating the multus networks. This design was rejected since new
-Jobs wouldn't be able to be created if the Kubernetes namespace were in "Terminating" state.
+When the Ceph-CSI plugin mounts or maps a block device it will use the network namespace of the
+Holder pod. This is achieved by using `nsenter` to enter the network namespace of the Holder pod.
+Some changes are expected as part of the mounting/mapping method in today's `csi-plugin` pod.
+Instead of invoking `rbd map ...`, invoke `nsenter --net=<net
+ns file of pause in long-running pod> rbd map ...` (and same for `mount -t ceph ...`).  The `csi-plugin`
+pod
+is already privileged, it already adds `CAP_SYS_ADMIN`. The only missing piece is to expose the host
+PID namespace with `hostPID: true` on the `csi-cephfsplugin` DaemonSet (RBD already has it).
+
+Cleaning up the symlink is not a requirement nor a big issue. Once the Holder contains stops, its
+network namespace goes away with it. So at worst, we end up with a broken symlink.
+
+When a new node is added, both DaemonSets are started, and the
+process described above occurs on the node.
+
+The initial implementation of this design is limited to supporting a single CephCluster with
+Multus.
+Until we stop using HostNetwork entirely it is impossible to support multiple CephClusters with and
```

Comment:
Do we need to support multiple cephclusters? Not sure we need to worry about it, or at least it's much lower priority.

Reply:
Rook supports deploying multiple clusters so it's natural to consider it :)